### PR TITLE
expose DNS entries for node port services

### PR DESCRIPTION
**Pull request by linki**
services exposed via node ports are ignored by `mate` at the moment.

For some setups it may make sense to have DNS records for those as well so they can be discovered just as any other exposed service. In this case the target IPs are the external IPs of the nodes. In `mate`'s design this means another producer that forwards endpoints for node port services. No change in the consumers is required if they can handle multiple target IPs for the same endpoint DNS name, see https://github.com/zalando-incubator/mate/pull/41 for making the Google consumer aware of that.

this leads to something like this:

```shell
$ dig +short prometheus-k8s.mate-test.example.org.
23.251.138.12
104.155.85.233
104.155.108.198
104.199.82.113
130.211.75.113
```

merge after https://github.com/zalando-incubator/mate/pull/41
merge after https://github.com/zalando-incubator/mate/pull/48
merge after https://github.com/zalando-incubator/mate/pull/51

### CODE REVIEW

**Comment by linki** [comment]
I made processing node port services configurable and it is disabled by default. Ready for review.

**Comment by linki** [comment]
or maybe wait until I rebased on the dependent PRs.

**Comment by ideahitme** [comment]
@linki conflict need to be resolved on this one

File: producers/kubernetes.go [hunk 1]
```diff
@@ -20,39 +20,52 @@ var kubernetesParams struct {
...
-	service *kubernetesServiceProducer
+	ingress   Producer
+	service   Producer
+	nodePorts Producer
 }
 
 func init() {
 	kingpin.Flag("kubernetes-server", "The address of the Kubernetes API server.").URLVar(&kubernetesParams.kubeServer)
 	kingpin.Flag("kubernetes-format", "Format of DNS entries, e.g. {{.Name}}-{{.Namespace}}.example.com").StringVar(&kubernetesParams.format)
+	kingpin.Flag("kubernetes-enable-node-port-services", "When true, generates DNS entries for type=NodePort services").BoolVar(&kubernetesParams.enableNodePorts)
```
Comment (ideahitme) [inline producers/kubernetes.go]: too verbose, maybe `kubernetes-track-node-ports` is enough ?

Comment (linki) [reply producers/kubernetes.go]: changed!

**Comment by ideahitme** [comment]
as the number of flags grows, we probably need to maintain the complete list somewhere, which can be as well helpful to track new features enabled through flags. E.g.: https://github.com/coreos/etcd/blob/master/Documentation/op-guide/configuration.md

**Comment by ideahitme** [comment]
@linki as discussed NodePort services are not supported with current AWS Consumer implementation, and this should be clearly documented. And possibly create an issue for future implementation :)

**Comment by linki** [comment]
@ideahitme documented the node port / aws incompatibility in the readme and code.

**Comment by ideahitme** [comment]
👍

**Comment by linki** [comment]
👍

**Pull request merged**